hypothesis-python/docs/reference/developer.rst

```diff
@@ -0,0 +1,29 @@
+Developer Reference
+===================
+
+.. warning::
+
+    This page documents APIs that Hypothesis makes available to developers building tools, libraries, or research on top of Hypothesis. It is not intended for users of Hypothesis.
+
+    While these APIs are mostly stable, they are considered internal, and are not subject to our standard :ref:`deprecation policy <deprecation-policy>`. We may make breaking changes to these APIs in minor or patch releases.
```

Comment:
I think I'd want to open with the stability policy, then who it's intended for, and keep that mostly positive.

> This page documents various internal interfaces, which are not subject to our standard :ref:`deprecation policy <deprecation-policy>`. Some are fairly stable; others are still experimental; in either case we might make breaking changes in minor or patch releases.
>
> We document these internals for the benefit of of people building tools, libraries, or research on top of Hypothesis.  If this includes you, please get in touch - we'd love to hear what you're doing, and explore more stable ways to support your use-case.

Reply:
double check this, I reworded a bit